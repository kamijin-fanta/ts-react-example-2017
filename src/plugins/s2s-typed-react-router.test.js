import plugin from './s2s-typed-react-router';
import pluginTester from 'babel-plugin-tester';

pluginTester({
  plugin,
  pluginOptions: {
  },
  snapshot: true,
  tests: [
    {
      title: 'single',
      code: `
        import { Path ,} from './lib';
        
        export const Routes: RoutesType = {
          Hello: Path({ url: '/', title: 'Top Page', component: Hello }),
          TodoView: Path({ url: '/todo/view/:page?', title: 'view todo', component: TodoView }),
          TodoEdit: Path({ url: '/todo/edit/:id', title: 'edit todo', component: TodoEdit }),
          TodoAdd: Path({ url: '/todo/add/:parent/:genre?', title: 'add todo', component: TodoAdd }),
        };

        /***** Do not edit below this line *****/
        export interface RoutesType {
          Hello: RoutePath<HelloRouteProps, EmptyParameterProps>;
          TodoView: RoutePath<TodoViewRouteProps, OptionalParameterProps<TodoViewRouteProps>>;
          TodoEdit: RoutePath<TodoEditRouteProps, ParameterProps<TodoEditRouteProps>>;
        }
        export interface HelloRouteProps {}
        interface TodoViewRouteProps {}
      `,
    },
  ],
});
