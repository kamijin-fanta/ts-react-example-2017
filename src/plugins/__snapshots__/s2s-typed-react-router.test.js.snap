// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single 1`] = `
"
import { Path ,} from './lib';

export const Routes: RoutesType = {
  Hello: Path({ url: '/', title: 'Top Page', component: Hello }),
  TodoView: Path({ url: '/todo/view/:page?', title: 'view todo', component: TodoView }),
  TodoEdit: Path({ url: '/todo/edit/:id', title: 'edit todo', component: TodoEdit }),
  TodoAdd: Path({ url: '/todo/add/:parent/:genre?', title: 'add todo', component: TodoAdd }),
};

export interface RoutesType {
  Hello: RoutePath<HelloRouteProps, EmptyParameterProps>;
  TodoView: RoutePath<TodoViewRouteProps, OptionalParameterProps<TodoViewRouteProps>>;
  TodoEdit: RoutePath<TodoEditRouteProps, ParameterProps<TodoEditRouteProps>>;
}
export interface HelloRouteProps {}
interface TodoViewRouteProps {}

      ↓ ↓ ↓ ↓ ↓ ↓

import { Path } from './lib';
export const Routes: RoutesType = {
  Hello: Path({
    url: '/',
    title: 'Top Page',
    component: Hello
  }),
  TodoView: Path({
    url: '/todo/view/:page?',
    title: 'view todo',
    component: TodoView
  }),
  TodoEdit: Path({
    url: '/todo/edit/:id',
    title: 'edit todo',
    component: TodoEdit
  }),
  TodoAdd: Path({
    url: '/todo/add/:parent/:genre?',
    title: 'add todo',
    component: TodoAdd
  })
};

/***** Do not edit below this line *****/
export interface RoutesType {
  Hello: RoutePath<HelloRouteProps, EmptyParameterProps>;
  TodoView: RoutePath<TodoViewRouteProps, OptionalParameterProps<TodoViewRouteProps>>;
  TodoEdit: RoutePath<TodoEditRouteProps, ParameterProps<TodoEditRouteProps>>;
  TodoAdd: RoutePath<TodoAddRouteProps, ParameterProps<TodoAddRouteProps>>;
}
export interface HelloRouteProps {}
export interface TodoViewRouteProps {
  page?: string;
}
export interface TodoEditRouteProps {
  id: string;
}
export interface TodoAddRouteProps {
  parent: string;
  genre?: string;
}
"
`;
